tests/helpers/test_state.py

```diff
@@ -186,9 +188,10 @@ def test_call_to_component(hass):
 async def test_as_number_states(hass):
     """Test state_as_number with states."""
     zero_states = (STATE_OFF, STATE_CLOSED, STATE_UNLOCKED,
-                   STATE_BELOW_HORIZON, STATE_NOT_HOME)
+                   STATE_BELOW_HORIZON, STATE_NOT_HOME, STATE_OFFLINE,
+                   STATE_UNAVAILABLE)
```

Comment:
unavailable is not 0, it means that we don't know the state. I don't think that we should assume that's 0.

Reply:
I was sort of ambivalent about this, and did ask on Discord without a response.

Reply:
Should Unavailable be handled in the prometheus integration rather than forcing it to zero here? I'm thinking along the lines of if the state is unavailable, simply don't even output the metric?

Reply:
Not outputting the metric seems the better approch. Being unavailable means that we don't have a metric after all.